### PR TITLE
Change some HTTP links to HTTPS

### PR DESCRIPTION
**Pull request by fuglede**
Several subdomains of scipy.org support SSL/TLS connections, so those may as well be used.

### CODE REVIEW

File: doc/source/tutorial/weave.rst [hunk 1]
```diff
@@ -2520,10 +2520,10 @@ cryptic error report due to the fact that ``stdio.h`` also defines the name.
 ``weave`` should probably try and handle this in some way. Other things...
 
 .. _PyInline: http://pyinline.sourceforge.net/
-.. _SciPy: http://www.scipy.org
+.. _SciPy: https://www.scipy.org
 .. _mingw32: http://www.mingw.org%3Ewww.mingw.org
 .. _NumPy: http://numeric.scipy.org/
```
Comment (larsmans) [inline doc/source/tutorial/weave.rst]: This one can be secured too.

File: doc/source/tutorial/weave.rst [hunk 2]
```diff
@@ -2520,10 +2520,10 @@ cryptic error report due to the fact that ``stdio.h`` also defines the name.
...
-.. _here: http://www.scipy.org/Weave
+.. _here: https://www.scipy.org/Weave
 .. _Python Cookbook: http://aspn.activestate.com/ASPN/Cookbook/Python
```
Comment (larsmans) [inline doc/source/tutorial/weave.rst]: And this one: it's now https://code.activestate.com/recipes/langs/python/

File: doc/source/tutorial/weave.rst [hunk 3]
```diff
@@ -2520,10 +2520,10 @@ cryptic error report due to the fact that ``stdio.h`` also defines the name.
...
 .. _binary_search():
     http://aspn.activestate.com/ASPN/Cookbook/Python/Recipe/81188
```
Comment (larsmans) [inline doc/source/tutorial/weave.rst]: https://code.activestate.com/recipes/81188/

File: doc/source/tutorial/weave.rst (hunk 1 above)
Comment (fuglede) [inline doc/source/tutorial/weave.rst]: This one was already secured in the original commit.

File: doc/source/tutorial/weave.rst (hunk 2 above)
Comment (fuglede) [inline doc/source/tutorial/weave.rst]: So was this one.

File: doc/source/tutorial/weave.rst (hunk 1 above)
Comment (larsmans) [inline doc/source/tutorial/weave.rst]: I'm sorry, I'm not quite sure what you mean.

Comment (larsmans) [inline doc/source/tutorial/weave.rst]: Also, this URL doesn't even work with HTTP...

Comment (fuglede) [inline doc/source/tutorial/weave.rst]: Sorry, I was looking at the wrong line. It looks like http://numeric.scipy.org/ and https://numeric.scipy.org/ serve different content though, so perhaps securing that one is not ideal?

**Comment by larsmans** [comment]
Merged, thanks!

**Pull request merged**